processes/processes.go

```diff
@@ -182,97 +183,63 @@ func (procPlg *procPlugin) CollectMetrics(metricTypes []plugin.MetricType) ([]pl
 	}
 
 	// init stateCount map with keys from States
+
 	for _, state := range States.Values() {
 		stateCount[state] = 0
 	}
+
 	// get all proc stats
 	stats, err := procPlg.mc.GetStats(procPath.(string))
 	if err != nil {
 		return nil, serror.New(err)
 	}
 	// calculate number of processes in each state
-	for _, instances := range stats {
-		for _, instance := range instances {
-			stateName := States[instance.State]
-			stateCount[stateName]++
-		}
-	}
 
-	// accumulate stats
-	accumulated := map[string]map[string]uint64{}
-	for procName, instances := range stats {
-		acc := setProcMetrics(instances)
-		accumulated[procName] = acc
+	for _, proc := range stats {
+		stateName := States[proc.State]
+		stateCount[stateName]++
 	}
 
-	duplicatedStats := map[metricKey]struct{}{}
-
 	// calculate metrics
 	for _, metricType := range metricTypes {
 		ns := metricType.Namespace()
-		isDynamic, _ := ns.IsDynamic()
+		if len(ns) < 4 {
+			return nil, serror.New(fmt.Errorf("Unknown namespace length. Expecting at least 4, is %d", len(ns)))
+		}
 
-		if isDynamic && len(ns) == 5 {
-			reqProcName := ns[3].Value
-			metricName := ns[4].Value
-			if reqProcName == "*" {
-				// ns[3] is wildcard for all processes
-				for procName, accStats := range accumulated {
-					key := metricKey{Proc: procName, Stat: metricName}
-					if _, ok := duplicatedStats[key]; ok {
-						continue
-					}
-					duplicatedStats[key] = struct{}{}
-
-					if val, ok := accStats[metricName]; ok {
-						// change dynamic namespace element value (= "*") to current process name
-						// whole namespace stays dynamic (ns[3].Name != "")
-						nuns := core.Namespace(append([]core.NamespaceElement{}, ns...))
-						nuns[3] = fillNsElement(&nuns[3], procName)
-						metric := plugin.MetricType{
-							Namespace_:   nuns,
-							Data_:        val,
-							Timestamp_:   time.Now(),
-							Unit_:        metricNames[metricName].unit,
-							Description_: metricNames[metricName].description,
+		isDynamic, _ := ns.IsDynamic()
+		if isDynamic {
+			//pid and name is dynamic = all
+			if ns[4].IsDynamic() && ns[5].IsDynamic() {
+				for _, proc := range stats {
+					procMetrics := setProcMetrics(proc)
+					for procMet, val := range procMetrics {
+						if procMet == ns[6].Value {
+							// change dynamic namespace element value (= "*") to current process name
+							// whole namespace stays dynamic (ns[3].Name != "")
+							nuns := core.Namespace(append([]core.NamespaceElement{}, ns...))
+							nuns[4] = fillNsElement(&nuns[3], proc)
```

Comment:
I sent update. Not tested (at least it compiles).
Strange thing .. it is definitely wrong, but I can't find a place where it went wrong :) And I've working module compiled for me.

I'll later this week confirm, if module is working properly. In general idea is that 
[4] = pid 
[5] = full process name.
I found some traces in history of cleaning up code. It might be, that some search/replace/refactoring did some bad joke.